From 8fa5fa50dbbb94f223635517079b4827513d88d8 Mon Sep 17 00:00:00 2001
From: Philipp Kerling <yol@casix.org>
Date: Wed, 15 Apr 2020 20:31:55 +0200
Subject: [PATCH] Fix missing cast to void* in %p format

POSIX says %p argument must be pointer to void. Newer libfmt seems to
have an issue with arbitary pointers, even failing at compile time, so
make sure the cast is there (fmt::ptr casts to void*).
---
 .../VideoPlayer/DVDCodecs/Video/AddonVideoCodec.cpp    | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/xbmc/cores/VideoPlayer/DVDCodecs/Video/AddonVideoCodec.cpp b/xbmc/cores/VideoPlayer/DVDCodecs/Video/AddonVideoCodec.cpp
index fdd6c1aeb354..c062daeb78c0 100644
--- a/xbmc/cores/VideoPlayer/DVDCodecs/Video/AddonVideoCodec.cpp
+++ b/xbmc/cores/VideoPlayer/DVDCodecs/Video/AddonVideoCodec.cpp
@@ -260,8 +260,14 @@ CDVDVideoCodec::VCReturn CAddonVideoCodec::GetPicture(VideoPicture* pVideoPictur
       }
     }
 
-    CLog::Log(LOGDEBUG, LOGVIDEO, "CAddonVideoCodec: GetPicture::VC_PICTURE with pts %llu %dx%d (%dx%d) %f %p:%d offset:%d,%d,%d, stride:%d,%d,%d", picture.pts, pVideoPicture->iWidth, pVideoPicture->iHeight, pVideoPicture->iDisplayWidth, pVideoPicture->iDisplayHeight, m_displayAspect,
-          picture.decodedData, picture.decodedDataSize, picture.planeOffsets[0], picture.planeOffsets[1], picture.planeOffsets[2], picture.stride[0], picture.stride[1], picture.stride[2]);
+    CLog::Log(LOGDEBUG, LOGVIDEO,
+              "CAddonVideoCodec: GetPicture::VC_PICTURE with pts {} {}x{} ({}x{}) {} {}:{} "
+              "offset:{},{},{}, stride:{},{},{}",
+              picture.pts, pVideoPicture->iWidth, pVideoPicture->iHeight,
+              pVideoPicture->iDisplayWidth, pVideoPicture->iDisplayHeight, m_displayAspect,
+              fmt::ptr(picture.decodedData), picture.decodedDataSize, picture.planeOffsets[0],
+              picture.planeOffsets[1], picture.planeOffsets[2], picture.stride[0],
+              picture.stride[1], picture.stride[2]);
 
     if (picture.width != m_width || picture.height != m_height)
     {
--- a/xbmc/windowing/X11/WinSystemX11.cpp	2020-02-29 04:56:32.000000000 -0600
+++ b/xbmc/windowing/X11/WinSystemX11.cpp	2020-04-26 17:19:00.784441374 -0500
@@ -1025,7 +1025,7 @@
 
   if(status == Success && items_read)
   {
-    CLog::Log(LOGDEBUG,"Window Manager Name: %s", data);
+    CLog::Log(LOGDEBUG,"Window Manager Name: %s", (char *) data);
   }
   else
     CLog::Log(LOGDEBUG,"Window Manager Name: ");
